pysrc/bytewax/execution.py

```diff
@@ -118,58 +365,71 @@ def spawn_cluster(
         worker_count_per_proc: Number of worker threads to start on
             each process.
 
+        cluster_should_stop: Returns if this cluster should gracefully
+            shutdown soon.
+
     """
     addresses = _gen_addresses(proc_count)
-    with Pool(processes=proc_count) as pool:
-        futures = [
-            pool.apply_async(
-                cluster_main,
-                (
-                    flow,
-                    input_builder,
-                    output_builder,
+
+    manager = Manager()
+    proc_raised_ex = manager.Event()
+
+    def proc_should_stop():
+        return cluster_should_stop() or proc_raised_ex.is_set()
+
+    ctx = multiprocessing.get_context("spawn")
```

Comment:
From everything I read, fork is tricky and can cause lots of mysterious errors. https://bugs.python.org/issue40379 I am not sure but using spawn might be necessary in complex situations like this.